share/revng/test/tests/abi/gather_symbols.py

```diff
@@ -23,67 +23,92 @@ def accept_arguments():
+        result = {}
+        for match in re.finditer(
+            b"\n0+([a-fA-F0-9]+)\s+"  # address (without leading 0s)
+            + b"(\w+)\s+"  # symbol type (local, global, etc)
+            + b"(?:(\w+)\s+|)"  # optional extra flags (weak, debug, etc)
+            + b"([\*\.\w]+)\s+"  # section symbol belongs to
+            + b"([a-fA-F0-9]+)\s+"  # either alignment or size of the symbol
+            + b"([\w\._-]+)+",  # name of the symbol
+            self.symbols,
+        ):
+            result[match[6].decode("utf-8")] = match[1].decode("utf-8")
+
+        return result
+
+    def find_section(self, name):
+        match = re.search(
+            b"\n\s+\d+\s+"  # index
+            + bytes(name, "utf-8")  # name
+            + b"\s+([a-fA-F0-9]+)\s+"  # size
+            + b"([a-fA-F0-9]+)\s+"  # vma
+            + b"([a-fA-F0-9]+)\s+"  # lma
+            + b"([a-fA-F0-9]+)\s+"  # offset within the file
+            + b"\d+\*\*\d+\s*",  # (uncaptured) alignment
+            self.sections,
         )
-    else:
-        raise Exception("unable to find a section: '" + name + "'")
-
+        if match:
+            return Section(
+                name,
+                match[1].decode("utf-8"),  # size
+                match[2].decode("utf-8"),  # vma
+                match[4].decode("utf-8"),  # offset
+            )
+        else:
+            raise Exception("unable to find a section: '" + name + "'")
+
+    def find_callsites(self, name):
+        result = []
+        for match in re.finditer(
+            b"\n\s+0*([a-fA-F0-9]+):"  # address (without leading 0s)
+            + b"[^<\n]*<"  # any number of non-`<` characters followed by `<`
+            + bytes(name, "utf-8")  # name
+            + b">\n"  # `>` character and a new line
+            + b"\s+0*([a-fA-F0-9]+)",  # address of the next instruction
+            self.disassembly,
+        ):
+            result.append((match[1].decode("utf-8"), match[2].decode("utf-8")))
+
+        return result
+
+
+class DumpbinOutputParser:
+    def __init__(self, sections, symbols, disassembly):
+        raise Exception("TODO")
 
-def find_callsites(name, data):
-    result = []
-    for match in re.finditer(
-        b"\n\s+0*([a-fA-F0-9]+):"  # address (without leading 0s)
-        + b"[^<\n]*<"  # any number of non-`<` characters followed by `<`
-        + bytes(name, "utf-8")  # name
-        + b">\n"  # `>` character and a new line
-        + b"\s+0*([a-fA-F0-9]+)",  # address of the next instruction
-        data,
-    ):
-        result.append((match[1].decode("utf-8"), match[2].decode("utf-8")))
 
-    return result
+def select_parser(sections, symbols, disassembly):
+    if re.match(b"^\n[^:]+:\s+file format elf.+\n", sections):
+        return ObjdumpOutputParser(sections, symbols, disassembly)
+    elif re.match(b"^\r\nDump of file .*\.exe\r\n\r\nPE signature found\r\n", sections):
+        return DumpbinOutputParser(sections, symbols, disassembly)
```

Comment:
Is the support to `dumpbin` new? If so, mention that in the commit message.

Reply:
There was a `TODO` in the file. But sure thing.

Reply:
I moved these lines to the next (`dumpbin`-focused) commit. Closing the thread.